csharp/src/Ice/UriParser.cs

```diff
@@ -215,15 +176,15 @@ internal static void RegisterCommon()
             return endpoint;
         }
 
-        private static (Uri Uri, string? AltEndpoint) InitialParse(
+        private static (Uri Uri, string? AltEndpoint, ProxyOptions ProxyOptions) InitialParse(
             string uriString,
-            ProxyOptions? proxyOptions,
+            bool pureEndpoints,
```

Comment:
what is `pureEndpoints` about, an endpoint that doesn't have identity or facet? a comment will make this clear